Sources/BezierSwift/MasterComponent/BezierDialog/BezierDialog.swift

```diff
@@ -6,15 +6,229 @@
+    self.priority = priority
+    self.title = title ?? ""
+    self.description = description ?? ""
+    self.buttonInfo = buttonInfo
+    self.dismissCompletion = dismissCompletion
+  }
+}
+
+public struct BezierDialogManager {
+  public static func show(with param: BezierDialogParam) {
+    BezierDialogSingleton.shared.viewModel.update(param: param)
+  }
+
+  public static func dismiss() {
+    BezierDialogSingleton.shared.viewModel.dismiss()
+  }
+}
+
+extension View {
+  @available(iOS 15.0, *)
+  public func initBezierDialog() -> some View {
+    self.modifier(BezierDialog())
+  }
 }
 
-struct BezierDialog_Previews: PreviewProvider {
-    static var previews: some View {
-      BezierDialog()
+class DialogViewModel: ObservableObject {
+  @Published private(set) var isPresented: Bool = false
+  
+  @Published var title: String = ""
+  @Published var description: String = ""
+  @Published var buttons: [BezierButton] = []
+  @Published var isButtonStackVertical = false
+  
+  var currentParamId: UUID = UUID()
+  var currentDismissCompletion: (() -> Void)?
+  var currentPriority: Int = Int.min
+  
+  func update(param: BezierDialogParam) {
+    guard param.priority > self.currentPriority else {
+      param.dismissCompletion?()
+      return
     }
+    
+    self.dismiss()
+    
+    DispatchQueue.main.asyncAfter(deadline: .now() + 0.3) { [weak self] in
+      guard let self = self else { return }
+      
+      self.currentParamId = param.id
+      self.currentDismissCompletion = param.dismissCompletion
+      self.currentPriority = param.priority
+      
+      self.title = param.title
+      self.description = param.description
+      
+      if let buttonInfo = param.buttonInfo {
+        switch buttonInfo {
+        case .vertical(let buttons):
+          self.buttons = buttons
+          self.isButtonStackVertical = true
+        case .horizontal(let buttons):
+          self.buttons = buttons
+          self.isButtonStackVertical = false
+        }
+      }
+    }
+    
+    self.isPresented = true
+  }
+  
+  func dismiss() {
+    self.currentDismissCompletion?()
+    
+    self.currentDismissCompletion = nil
+    self.currentParamId = UUID()
+    self.currentPriority = Int.min
+    
+    self.isPresented = false
+  }
+}
+
+class BezierDialogSingleton {
+  var viewModel: DialogViewModel
+  
+  static let shared = BezierDialogSingleton()
+
+  init() {
+    self.viewModel = DialogViewModel()
+  }
 }
+
+@available(iOS 15.0, *)
+struct BezierDialog: ViewModifier, Themeable {
+  @Environment(\.colorScheme) public var colorScheme
+  
+  @StateObject private var viewModel: DialogViewModel = BezierDialogSingleton.shared.viewModel
+  
+  
+  func body(content: Content) -> some View {
+    
+    content
+      .overlay {
+        if viewModel.isPresented {
+          self.palette(.bgtxtAbsoluteBlackLighter)
+            .edgesIgnoringSafeArea(.all)
```

Comment:
```suggestion
            .edgesIgnoringSafeArea(.all)
            .frame(maxWidth: .infinity, maxHeight: .infinity)
```
좀더 명시적으로 프레임 잡아주면 Spacer는 지워도 될 것 같습니다.